src/lib.js

```diff
@@ -2396,6 +2397,54 @@ export const getStats = async (
     misc.claimed_items = hypixelProfile.claimed_items;
   }
 
+  const response = await axios("https://api.hypixel.net/resources/achievements");
```

Comment:
Don't really like this, it gets called every time a profile loads... we should fetch it and cache it in db maybe like... every hour or so

Reply:
Agreed. But where sould I cache it? New table one value and put the whole list in there? How does that work? I've never used db.collcetion so far...

Reply:
Me neither honestly, I would look at how we store bazaar data (that gets refreshed every 30min or so I think) and do it in the same way

Reply:
I've tried but the query stays empty. There has to be some place where these are properly initialized - but the lines I set in init-collections.js seem not to work properly either:
´´´
  db.collection("achievements_tiered").createIndex({ key: "text" }, { unique: true }),
  db.collection("achievements_one_time").createIndex({ key: "text" }, { unique: true }),
´´´
```
let tieredAchievements = await db.collection("achievements_tiered").find().toArray();
  let oneTimeAchievements = await db.collection("achievements_one_time").find().toArray();
  console.log(tieredAchievements);
  console.log(oneTimeAchievements);
  if (!tieredAchievements || tieredAchievements == [] || !oneTimeAchievements || oneTimeAchievements == []) {
    //||
    //+new Date() - tieredAchievements?.last_updated > 3600 * 1000 ||
    //+new Date() - oneTimeAchievements?.last_updated > 3600 * 1000
    const response = await axios("https://api.hypixel.net/resources/achievements");
    let skyblock_achievements = response.data.achievements.skyblock;
    let tieredList = skyblock_achievements.tiered;
    let oneTimeList = skyblock_achievements.one_time;

    for (const key in tieredList) {
      const tmp = tieredList[key];
      await db
        .collection("achievements_tiered")
        .updateOne(
          { key },
          { $set: { name: tmp.name, description: tmp.description, tiers: tmp.tiers } },
          { upsert: true }
        );
    }
    for (const key in oneTimeList) {
      const tmp = oneTimeList[key];
      await db
        .collection("achievements_one_time")
        .updateOne(
          { key },
          { $set: { name: tmp.name, description: tmp.description, points: tmp.points, amount: tmp.amount } },
          { upsert: true }
        );
    }
    tieredAchievements = await db.collection("achievements_tiered").find().toArray();
    oneTimeAchievements = await db.collection("achievements_one_time").find().toArray();
    console.log(tieredAchievements);
    console.log(oneTimeAchievements);
  }
```

Reply:
Sorry for directly commiting it to your repo, but I hope this is what you wanted.

Reply:
> Sorry for directly commiting it to your repo, but I hope this is what you wanted.

On the contrary - one of the reasons I allow "edits by mantainers". It was a little bit out of my reach of my current knowledge about this project, javascript and mongodb. Thank you for your commit to my repo.